### PR TITLE
FYST-791 ID health insurance premium subtraction

### PR DESCRIPTION
**Pull request by arinchoi03**
## Link to pivotal/JIRA issue
[FYST-791](https://codeforamerica.atlassian.net/browse/FYST-791)
## Is PM acceptance required? (delete one)
- Yes - don't merge until JIRA issue is accepted!

**Reminder**: merge main into this branch and get green tests before merging to main
## What was done?
- Added Health Insurance Premium page & reflected that in the ID final review screen
- Also fixed a copy in ID Sales/Use Tax that looked wrong
## How to test?
- Describe the testing approach taken to verify the changes, including:
  - Unit/feature specs written & manually tested as well
  - Any fixture can be used, just need to input the health insurance paid amount value and see it be reflected on review page (rounded to 2nd decimal place) & XML/PDF (rounded to integer)
   - Check copy singular / plural on the question page:
     - single filer: "you/yourself"
     -  mfj / single filer with dependents: "you and your household/yourself an your household"
- Use deployed heroku env
## Screenshots (for visual changes)
Screenshot 2024-10-29 at 4 05 39 PM
Screenshot 2024-10-29 at 4 05 43 PM
Screenshot 2024-10-29 at 4 05 55 PM

[FYST-791]: https://codeforamerica.atlassian.net/browse/FYST-791?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/forms/state_file/id_health_insurance_premium_form.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module StateFile
+  class IdHealthInsurancePremiumForm < QuestionsForm
+    set_attributes_for :intake, :has_health_insurance_premium, :health_insurance_paid_amount
+
+    validates :has_health_insurance_premium, inclusion: { in: %w[yes no], message: :blank }
+    validates :health_insurance_paid_amount,
+      presence: true,
+      numericality: {
+        allow_blank: true,
+        greater_than_or_equal_to: 0,
```
Comment (arinchoi03) [inline app/forms/state_file/id_health_insurance_premium_form.rb:10]: open question whether we should allow `0` value for this amount, but should be ok to review

File: app/models/state_file_base_intake.rb [hunk 2]
```diff
@@ -202,6 +202,10 @@ def filer_count
     filing_status_mfj? ? 2 : 1
   end
 
+  def household_count
+    filer_count + dependents.size
+  end
```
Comment (arinchoi03) [inline app/models/state_file_base_intake.rb:207]: mike is implementing this on another PR but don't know who will get to merge it first

File: app/forms/state_file/id_health_insurance_premium_form.rb (hunk 1 above)
Comment (arinchoi03) [reply app/forms/state_file/id_health_insurance_premium_form.rb:10]: ok to allow 0 here!
>1- it’s possible that the filer may think that they paid Health Insurance Premiums when they initially respond to the question…..but as they more carefully review all of the supporting text in the reveals describing which Health Insurance payments actually qualify for this subtraction they may realize that the Health Insurance Premium payments that they made don’t actually qualify. So allowing them to enter 0 (and filling the PDF field and updating the state XML field with it) is the behavior that we are looking for here.  

https://codeforamerica.atlassian.net/browse/FYST-791?focusedCommentId=12716

File: app/views/state_file/questions/id_sales_use_tax/edit.html.erb [hunk 3]
```diff
@@ -11,7 +11,6 @@
           <%=
             f.cfa_radio_set(
               :has_unpaid_sales_use_tax,
-              label_text: t('.unpaid_sales_use_tax_label_html'),
               collection: [
```
Comment (tahsinaislam) [inline app/views/state_file/questions/id_sales_use_tax/edit.html.erb:14]: why was this taken out?

File: app/forms/state_file/id_health_insurance_premium_form.rb (hunk 1 above)
Comment (tahsinaislam) [reply app/forms/state_file/id_health_insurance_premium_form.rb:10]: if they answer no for health insurance premium question, should we still be adding 0 for the xml and pdf?

File: spec/features/state_file/complete_intake_spec.rb [hunk 4]
```diff
@@ -492,12 +492,21 @@
...
+      # Sales/Use Tax
       expect(page).to have_text I18n.t('state_file.questions.id_sales_use_tax.edit.title', year: MultiTenantService.statefile.current_tax_year)
       choose I18n.t("general.affirmative")
       fill_in 'state_file_id_sales_use_tax_form_total_purchase_amount', with: "290"
+      click_on I18n.t("general.continue")
 
       # ID Review page
-      click_on I18n.t("general.continue")
+      expect(page).to have_text "Idaho Health Insurance Premium Subtraction"
+      expect(page).to have_text "$1,234.60"
```
Comment (tahsinaislam) [inline spec/features/state_file/complete_intake_spec.rb:510]: hmm wondering if going forward we should all add the review page content

**Review by tahsinaislam — APPROVED** [review]
some questions i was wondering but otherwise looks amazing

File: app/forms/state_file/id_health_insurance_premium_form.rb (hunk 1 above)
Comment (arinchoi03) [reply app/forms/state_file/id_health_insurance_premium_form.rb:10]: ooh i'll ask this

File: app/views/state_file/questions/id_sales_use_tax/edit.html.erb (hunk 3 above)
Comment (arinchoi03) [reply app/views/state_file/questions/id_sales_use_tax/edit.html.erb:14]: there is not supposed to be a copy above this Yes/No question -- according to figma. I had included the same text twice & noticed it was different from the figma, so I removed it

File: spec/features/state_file/complete_intake_spec.rb (hunk 4 above)
Comment (arinchoi03) [reply spec/features/state_file/complete_intake_spec.rb:510]: yes i think we should

**Pull request merged**